### PR TITLE
fix: correctly pass resolvedBaseUrl for Next.js 13.2.0+

### LINKED ISSUE #25951: Component testing broken with NextJS 13.2.1
### Current behavior

Component tests run fine on NextJS v13.1.1, but fail to compile on v13.2.1

The error received is unable to find module, where the module resolution relies on the `tsconfig` basePath setting in `/cypress/tsconfig.json`

Component test cypress config settings:

```
component: {
    devServer: {
      framework: 'next',
      bundler: 'webpack',
    },
}
```

### Desired behavior

Component tests should compile and run using the `tsconfig.json` settings.

Possibly related NextJS change/update: https://github.com/vercel/next.js/pull/45455

### Test code to reproduce

Run any component test that uses a custom cypress `tsconfig` with NextJS v13.2.1

### Cypress Version

latest

### Node version

v16.3

### Operating System

MacOSX And Github Actions Linux

### Debug Logs

_No response_

### Other

_No response_

### PR DESCRIPTION
**Pull request by lmiller1990**
- Closes https://github.com/cypress-io/cypress/issues/25951

### Additional details

As of v13.1.7-canary.7, Next.js projects setting `baseUrl` to anything other than the default `"."` would not correctly resolve modules. 

Diff: https://github.com/vercel/next.js/compare/v13.1.7-canary.6...v13.1.7-canary.7. Lots of files but mostly static content. [Here's the exact line that broke us](https://github.com/vercel/next.js/compare/v13.1.7-canary.6...v13.1.7-canary.7#diff-0260dbfc6a047be5877f047fb9f7205b57920663963f0dadd6fffe912b6d386bR491-R495).

```js
getBaseWebpackConfig(this.dir, {
  ...commonWebpackOptions,
  compilerType: COMPILER_NAMES.client,
  entrypoints: entrypoints.client,
  ...info // <====== new in 13.2.0
})
```

`info` contains `info.resolvedBaseUrl`, which we are currently not passing. I updated our nextHandler to include the `resolvedBaseUrl`, and it works. I added a regression test, too.

### Steps to test

You could simply revert the fix I made and run the system test - it will fail. Alternatively, you could make a new Next project, change `baseUrl` to `src`, then update some imports to remove `src` - eg `import 'src/styles/global.css'` would become `import 'styles/global.css'`, which is allowed since you've now got `"baseUrl": "src"`.

### How has the user experience changed?

It respects `baseUrl` and works as expected.

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: npm/webpack-dev-server/src/helpers/nextHandler.ts [hunk 1]
```diff
@@ -196,8 +196,10 @@ async function loadWebpackConfig (devServerConfig: WebpackDevServerConfig): Prom
       compilerType: 'client',
       // Required for Next.js > 13
       hasReactRoot: reactVersion === 18,
-      // Required for Next.js > 13.2.1 to respect TS/JS config
+      // Required for Next.js > 13.2.0 to respect TS/JS config
       jsConfig: jsConfigResult.jsConfig,
+      // Required for Next.js > 13.2.0 to respect tsconfig.compilerOptions.baseUrl
+      resolvedBaseUrl: jsConfigResult.resolvedBaseUrl,
```
Comment (lmiller1990) [inline npm/webpack-dev-server/src/helpers/nextHandler.ts:217]: This is the fix. We could do `...jsConfigResult`, but we don't know exactly what else is there (can change at any time), I think this is probably more explicit and maintainable.

**Review by mike-plummer — APPROVED** [review]
Nice work tracking this down ! 🥳 

Two thoughts on the changelog entry:
1. Should it reference "in Component Testing projects" just to be more explicit?
2. #25951 is already referenced in the 12.8.0 changelog - should we add wording around "completed an edge case not covered in our original fix" or something to that effect?

**Review by jordanpowell88 — APPROVED** [review]
LGTM. Great job tracking this down @lmiller1990

**Comment by lmiller1990** [comment]
@mike-plummer good idea, I updated the changelog.

**Pull request merged**